api/src/main/java/javax/servlet/http/HttpServlet.java

```diff
@@ -367,76 +356,44 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
         Method[] methods = getAllDeclaredMethods(this.getClass());
 
-        boolean ALLOW_GET = false;
-        boolean ALLOW_HEAD = false;
-        boolean ALLOW_POST = false;
-        boolean ALLOW_PUT = false;
-        boolean ALLOW_DELETE = false;
-        boolean ALLOW_TRACE = true;
-        boolean ALLOW_OPTIONS = true;
-
-        for (int i = 0; i < methods.length; i++) {
-            String methodName = methods[i].getName();
-
-            if (methodName.equals("doGet")) {
-                ALLOW_GET = true;
-                ALLOW_HEAD = true;
-            } else if (methodName.equals("doPost")) {
-                ALLOW_POST = true;
-            } else if (methodName.equals("doPut")) {
-                ALLOW_PUT = true;
-            } else if (methodName.equals("doDelete")) {
-                ALLOW_DELETE = true;
-            }
-
-        }
-
-        // we know "allow" is not null as ALLOW_OPTIONS = true
+        // we know "allow" is not null OPTIONS method are allowed
         // when this method is invoked
         StringBuilder allow = new StringBuilder();
-        if (ALLOW_GET) {
-            allow.append(METHOD_GET);
-        }
-        if (ALLOW_HEAD) {
-            if (allow.length() > 0) {
-                allow.append(", ");
-            }
-            allow.append(METHOD_HEAD);
-        }
-        if (ALLOW_POST) {
-            if (allow.length() > 0) {
-                allow.append(", ");
-            }
-            allow.append(METHOD_POST);
-        }
-        if (ALLOW_PUT) {
-            if (allow.length() > 0) {
-                allow.append(", ");
-            }
-            allow.append(METHOD_PUT);
-        }
-        if (ALLOW_DELETE) {
-            if (allow.length() > 0) {
-                allow.append(", ");
-            }
-            allow.append(METHOD_DELETE);
-        }
-        if (ALLOW_TRACE) {
-            if (allow.length() > 0) {
-                allow.append(", ");
-            }
-            allow.append(METHOD_TRACE);
-        }
-        if (ALLOW_OPTIONS) {
-            if (allow.length() > 0) {
-                allow.append(", ");
+
+        for (Method value : methods) {
+            String methodName = value.getName();
+
+            switch (methodName) {
+                case "doGet":
+                    appendAllowedMethod(allow, METHOD_GET);
+                    appendAllowedMethod(allow, METHOD_HEAD);
+                    break;
+                case "doPost":
+                    appendAllowedMethod(allow, METHOD_POST);
+                    break;
+                case "doPut":
+                    appendAllowedMethod(allow, METHOD_PUT);
+                    break;
+                case "doDelete":
+                    appendAllowedMethod(allow, METHOD_DELETE);
+                    break;
             }
-            allow.append(METHOD_OPTIONS);
         }
 
+        // TRACE method is allowed since it used in debugging
+        appendAllowedMethod(allow, METHOD_TRACE);
```

Comment:
I wish there was a way to not have `TRACE` show up in the default `OPTIONS` `Allow` method if the user so desires to configure it that way.

Something at the ServletContext level tho.
To compliment those users that have Constraint mappings to disable TRACE, so should this default `OPTIONS` method not even show `TRACE` if it's been disabled at the higher level.